src/platform_impl/macos/event.rs

```diff
@@ -113,6 +115,14 @@ pub(crate) fn create_key_event(
     let scancode = unsafe { ns_event.keyCode() };
     let mut physical_key = key_override.unwrap_or_else(|| scancode_to_physicalkey(scancode as u32));
 
+    // NOTE: The logical key should heed both SHIFT and ALT if possible.
+    // For instance:
+    // * Pressing the A key: logical key should be "a"
+    // * Pressing SHIFT A: logical key should be "A"
+    // * Pressing CTRL SHIFT A: logical key should also be "A"
+    // This is not easy to tease out of `NSEvent`, but we do our best.
+
+    // `text_with_all_modifiers` is unreliable when ctrl or cmd is pressed
```

Comment:
This looks out of place here, it matters when you do `!has_crtl && !has_cmd` and it already has similar comment.